<?xml version="1.0"?>

<launch>
  <arg name="uri" default="usb://0" />

  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
  </include>

  <group ns="crazyflie">

    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri)" />
      <param name="tf_prefix" value="crazyflie" />
      <param name="enable_logging" value="True" />
      <param name="enable_logging_temperature" value="False" />
      <param name="enable_logging_magnetic_field" value="False" />
      <param name="enable_logging_pressure" value="False" />
      <param name="enable_logging_imu" value="True" />
      <param name="enable_logging_battery" value="False" />
      <rosparam>
        genericLogTopics: ["log_pos" , "log_rpy" , "log_ranges" , "log_sensors" ]
        genericLogTopicFrequencies: [100 , 100 , 100 , 100]
        genericLogTopic_log_pos_Variables: ["stateEstimate.x", "stateEstimate.y", "stateEstimate.z"]
        genericLogTopic_log_rpy_Variables: ["stabilizer.roll" , "stabilizer.pitch" , "stabilizer.yaw"]
        genericLogTopic_log_ranges_Variables: ["range.front", "range.back", "range.up", "range.left", "range.right" , "range.zrange" ]
        genericLogTopic_log_sensors_Variables: ["motion.deltaX" , "motion.deltaY" , "motion.outlierCount" ]
      </rosparam>
    </node>


  </group>
</launch>
